database/src/desktop/util_desktop_macos.mm

```diff
@@ -17,11 +17,47 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include "database/src/desktop/util_desktop.h"
+#include <string>
 
 namespace firebase {
 namespace database {
 namespace internal {
 
+// Split a string based on specified character delimiter into constituent parts
+static std::vector<std::string> split_string(const std::string& s,
+                                             const char delimiter='/') {
+  size_t pos = 0;
+  // This index is used as the starting index to search the delimiters from.
+  size_t delimiter_search_start = 0;
+  // Skip any leading delimiters
+  while(s[delimiter_search_start] == delimiter) {
+    delimiter_search_start++;
+  }
+
+  std::vector<std::string> split_parts;
+  size_t len = s.size();
+  // Cant proceed if input string consists of just delimiters
+  if (pos >= len) {
+    return split_parts;
+  }
+
+  while( (pos = s.find(delimiter, delimiter_search_start)) != std::string::npos) {
+    split_parts.push_back(s.substr(delimiter_search_start, pos-delimiter_search_start));
+
+    while(s[pos] == delimiter && pos<len) {
+      pos++;
+      delimiter_search_start = pos;
+    }
+  }
+
+  // If the input string doesn't end with a delimiter we need to push the last
+  // token into our return vector
+  if (delimiter_search_start != len)
+    split_parts.push_back(s.substr(delimiter_search_start, len-delimiter_search_start));
```

Comment:
recommend putting this in brackets, like the if (pos >= len) above.

Reply:
Done!